core/iwasm/common/wasm_runtime_common.c

```diff
@@ -3934,6 +3934,184 @@ wasm_runtime_get_export_type(WASMModuleCommon *const module, int32 export_index,
 #endif
 }
 
+WASMFuncType *
+wasm_runtime_get_export_function(struct WASMModuleCommon *const module,
+                                 const char *name)
+{
```

Comment:
🤔 maybe there is another way to get `WASMFuncType`:

- build a `WASMExport` or use `wasm_runtime_get_export_type`(may need some modification)
- then use `wasm_runtime_get_export_func_type`

Reply:
There is already wasm_export_type_t/wasm_import_type_t, and the input of wasm_runtime_get_export_type/wasm_runtime_get_export_type is `uint32 export_index`/`uint32 import_index`, but here it wants to retrieve the func type info according to the name, we cannot pass name to the two APIs. One possible way is to add new APIs, like:
```C
WASM_RUNTIME_API_EXTERN void
wasm_runtime_get_export_type_by_name(const wasm_module_t module, const char *name,
                             wasm_export_type *export_type);
WASM_RUNTIME_API_EXTERN void
wasm_runtime_get_import_type_by_name(const wasm_module_t module, const char *name,
                             wasm_import_type *import_type);
```
And extend the structure wasm_import_type and wasm_export_type:
```C
typedef struct wasm_import_type {
    const char *module_name;
    const char *name;
    wasm_import_export_kind_t kind;
    bool linked;
    union {
        wasm_func_type_t func_type;  // func type when the import is a function
        ...  // other info for other import kind
    } u;
} wasm_import_type;

typedef struct wasm_export_type {
    const char *name;
    wasm_import_export_kind_t kind;
    union {
        wasm_func_type_t func_type;  // func type when the import is a function
        ...  // other info for other export kind
} wasm_export_type;
```
And then we get the param/result info from func_type in the result of wasm_export_type and wasm_import_type by using wasm_func_type_get_param_types and wasm_func_type_get_result_types.

Reply:
BTW, @bnason-nf, could you change `wasm_export_type` and `wasm_import_type` to `wasm_export_type_t` and `wasm_import_type_t`? We usually added `_t` in the types declared in wasm_export.h.

Reply:
Hi @wenyongh, that seems to contradict what @lum1n0us said here:
https://github.com/bytecodealliance/wasm-micro-runtime/pull/3330#discussion_r1569823168